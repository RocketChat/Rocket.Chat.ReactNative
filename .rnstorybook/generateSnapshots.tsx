import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
