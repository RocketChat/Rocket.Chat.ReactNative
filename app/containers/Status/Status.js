import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { STATUS_COLORS, themes } from '../../constants/colors';

const Status = React.memo(({
	status, size, style, theme
}) => (
	<View
		style={
			[
				style,
				{
					borderRadius: size,
					width: size,
					height: size,
					backgroundColor: STATUS_COLORS[status],
					borderColor: themes[theme].backgroundColor
				}
			]}
	/>
));
Status.propTypes = {
	status: PropTypes.string,
	size: PropTypes.number,
	style: PropTypes.any,
	theme: PropTypes.string
};
Status.defaultProps = {
	status: 'offline',
	size: 16,
	theme: 'light'
};

export default Status;
