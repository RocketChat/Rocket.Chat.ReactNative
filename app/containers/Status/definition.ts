import { TextProps } from 'react-native';

import { TUserStatus } from '../../definitions';

export interface IStatus extends TextProps {
	id: string;
	size: number;
	status: TUserStatus;
}
