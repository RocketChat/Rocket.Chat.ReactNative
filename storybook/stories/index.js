/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
