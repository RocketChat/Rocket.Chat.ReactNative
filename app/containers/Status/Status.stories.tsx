import React from 'react';

import Status from './Status';

export default {
	title: 'Status'
};

export const All = () => (
	<>
		<Status status='online' />
		<Status status='busy' />
		<Status status='away' />
		<Status status='loading' />
		<Status status='disabled' />
		<Status status='offline' />
		<Status />
		<Status status='online' size={60} />
	</>
);
