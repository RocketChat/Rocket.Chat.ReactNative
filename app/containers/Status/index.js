import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Status from './Status';
import { withTheme } from '../../theme';

class StatusContainer extends React.PureComponent {
	static propTypes = {
		style: PropTypes.any,
		size: PropTypes.number,
		status: PropTypes.string,
		theme: PropTypes.string
	};

	static defaultProps = {
		size: 16
	}

	render() {
		const {
			style, size, status, theme
		} = this.props;
		return <Status size={size} style={style} status={status} theme={theme} />;
	}
}

const mapStateToProps = (state, ownProps) => ({
	status: state.meteor.connected ? state.activeUsers[ownProps.id] : 'offline'
});

export default connect(mapStateToProps)(withTheme(StatusContainer));
