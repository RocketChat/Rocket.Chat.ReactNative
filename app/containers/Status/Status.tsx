import React from 'react';
import { StyleProp, TextStyle } from 'react-native';

import { CustomIcon, IconSet, TIconsName } from '../CustomIcon';
import { STATUS_COLORS } from '../../lib/constants';
import { IStatus } from './definition';

const Status = React.memo(({ style, status = 'offline', size = 32, ...props }: Omit<IStatus, 'id'>) => {
	const name: TIconsName = `status-${status}`;
	const isNameValid = IconSet.hasIcon(name);
	const iconName = isNameValid ? name : 'status-offline';
	const calculatedStyle: StyleProp<TextStyle> = [
		{
			width: size,
			height: size,
			textAlignVertical: 'center'
		},
		style
	];

	return (
		<CustomIcon
			style={calculatedStyle}
			size={size}
			name={iconName}
			color={STATUS_COLORS[status] ?? STATUS_COLORS.offline}
			{...props}
		/>
	);
});

export default Status;
