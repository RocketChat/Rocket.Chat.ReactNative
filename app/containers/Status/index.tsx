import React from 'react';

import { TUserStatus } from '../../definitions';
import Status from './Status';
import { IStatus } from './definition';
import { useAppSelector } from '../../lib/hooks';

const StatusContainer = ({ id, style, size = 32, ...props }: Omit<IStatus, 'status'>): React.ReactElement => {
	const status = useAppSelector(state => {
		if (state.settings.Presence_broadcast_disabled) {
			return 'disabled';
		}
		if (state.meteor.connected) {
			return state.activeUsers[id] && state.activeUsers[id].status;
		}
		return 'loading';
	}) as TUserStatus;
	return <Status size={size} style={style} status={status} {...props} />;
};

export default StatusContainer;
