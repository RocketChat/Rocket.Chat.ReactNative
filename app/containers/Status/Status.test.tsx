import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './Status.stories';

generateSnapshots(stories);
