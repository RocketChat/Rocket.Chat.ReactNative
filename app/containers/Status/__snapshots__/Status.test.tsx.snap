// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: All should match snapshot 1`] = `
[
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#158D65",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#D40C26",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#AC892F",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#6C727A",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#F38C39",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#6C727A",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#6C727A",
          "fontSize": 64,
        },
        [
          {
            "lineHeight": 64,
          },
          [
            {
              "height": 64,
              "textAlignVertical": "center",
              "width": 64,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#158D65",
          "fontSize": 120,
        },
        [
          {
            "lineHeight": 120,
          },
          [
            {
              "height": 120,
              "textAlignVertical": "center",
              "width": 120,
            },
            undefined,
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>,
]
`;
